aten/src/ATen/native/Linear.cpp

```diff
@@ -641,9 +642,26 @@ Tensor tensordot(const Tensor& input1, const Tensor& input2, IntArrayRef dims1,
 }
 
 Tensor &tensordot_out(const Tensor& input1, const Tensor& input2, IntArrayRef dims1, IntArrayRef dims2, Tensor& result) {
-  result.copy_(at::native::tensordot(input1, input2, dims1, dims2));
+  Tensor result_tmp = at::native::tensordot(input1, input2, dims1, dims2);
+  auto result_dtype = result_tmp.scalar_type();
+  auto output_tensor_dtype = result.scalar_type();
+  auto output_device = result.device();
+  auto input_device = input1.device();
+  // check if the input & output tensors are on the same device.
+  TORCH_CHECK(
+    output_device == input_device,
+    "tensordot_out: Expected the output and input tensors to be on the "
+    "same device, but got output on ", output_device, " and inputs on ",
+    input_device);
+  // check if the computed result can be safely copied to the output tensor.
+  bool can_cast = c10::canCast(result_dtype, output_tensor_dtype);
```

Comment:
Use of canCast is great here. I wonder if we should think about including this check in resize_output() in the future. For example, maybe resize_output() could have a signature like: result, shape, computation dtype.

Reply:
Thanks, that'd be much cleaner!  `Linear.cpp` would use only two functions defined in `LinearAlgebraUtils.h`, so I had [thought](https://github.com/pytorch/pytorch/pull/56286#discussion_r615182402) it'd be better to discuss.